### ISSUE TITLE
Borders of canvas turn black when zooming into multi-resolution image

### ISSUE THREAD

**Issue by VolkerH** [napari/napari#4056]
## 🐛 Bug

This recording demonstrates the bug.

When zooming into a multi-resolution image, strips along the canvas borders turn black, although there should be valid pixel data.
The exact width and position of the black  edge strips change with the zoom level, creating a very jumpy impression while zooming.

![black_borders_multires](https://user-images.githubusercontent.com/2210044/152511272-c4862198-d3b6-4015-b1ff-fddc67a8fddf.gif)

## To Reproduce

Unfortunately I don't have a stand-alone repdroducible example. The error occurs while I have a multi-resolution Zarr open in a layer that also has a non-identity affine attached. In addition there are also shapes and points layers (this may be entirely irrelevant, but maybe this is still related to #3863, so I'll mention it here) . 

## Expected behavior

Valid pixel data displayed at borders when zooming in.

## Environment
napari: 0.4.13.dev156+g7341615b
Platform: Linux-5.13.0-27-generic-x86_64-with-glibc2.17
System: Ubuntu 20.04.3 LTS
Python: 3.8.12 (default, Oct 12 2021, 13:49:34) [GCC 7.5.0]
Qt: 5.15.2
PyQt5: 5.15.6
NumPy: 1.21.5
SciPy: 1.7.3
Dask: 2021.06.2
VisPy: 0.9.4

OpenGL:
- GL version: 4.6.0 NVIDIA 460.91.03
- MAX_TEXTURE_SIZE: 32768

Screens:
- screen 1: resolution 3840x2160, scale 1.0

Plugins:
- console: 0.0.4
- napari-ome-zarr: 0.2.1.dev6+gb626109
- napari-properties-viewer: 0.0.2
- napari-spacem-layout-io: 0.0.1
- scikit-image: 0.4.13.dev156+g7341615b

**Comment by jni** [comment]
Good point! I guess we need to draw all the way up to corner_pixels + [-1, +1] in data coordinates, rather than just to corner_pixels.
